azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -37,29 +37,29 @@ public void onExecutionEvent(final ExecutableFlow flow, final String action) {
 
   /**
    * A new execution will be dispatched based on the original ExecutableFLow
-   * @param exFlow original ExecutableFlow in EXECUTION_STOPPED state
+   * @param originalExFlow original ExecutableFlow in EXECUTION_STOPPED state
    */
-  private void restartExecutableFlow(final ExecutableFlow exFlow) {
+  void restartExecutableFlow(final ExecutableFlow originalExFlow) {
     // Enable restartability for containerized execution
-    if (exFlow.getDispatchMethod() != DispatchMethod.CONTAINERIZED) return;
+    if (originalExFlow.getDispatchMethod() != DispatchMethod.CONTAINERIZED) return;
 
     // Create a new ExecutableFlow based on existing flow in EXECUTION_STOPPED state
     final Project project;
     final Flow flow;
     try {
-      project = FlowUtils.getProject(this.projectManager, exFlow.getProjectId());
-      flow = FlowUtils.getFlow(project, exFlow.getFlowId());
+      project = FlowUtils.getProject(this.projectManager, originalExFlow.getProjectId());
+      flow = FlowUtils.getFlow(project, originalExFlow.getFlowId());
     } catch (final RuntimeException e) {
       logger.error(e.getMessage());
       return;
     }
-    final ExecutableFlow executableFlow =
+    final ExecutableFlow retryExFlow =
         this.executorManagerAdapter.createExecutableFlow(project, flow);
-    executableFlow.setSubmitUser(exFlow.getSubmitUser());
-    executableFlow.setExecutionSource(Constants.EXECUTION_SOURCE_ADHOC);
-    executableFlow.setUploadUser(project.getUploadUser());
+    retryExFlow.setSubmitUser(originalExFlow.getSubmitUser());
+    retryExFlow.setExecutionSource(Constants.EXECUTION_SOURCE_ADHOC);
```

Comment:
I don't think the execution_source should be ad-hoc. It should be something like "auto_retry". Adhoc means user adhoc execution.

Reply:
+1

Reply:
good point, let me add a new constant